code/game/machinery/computer/buildandrepair.dm

```diff
@@ -1,271 +1,394 @@
+
+			if(istype(P, /obj/item/weapon/screwdriver) && circuit)
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+				user << "<span class='notice'>You screw the circuit board into place.</span>"
+				state = 2
+				icon_state = "2"
+			if(istype(P, /obj/item/weapon/crowbar) && circuit)
+				playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
+				user << "<span class='notice'>You remove the circuit board.</span>"
+				state = 1
+				icon_state = "0"
+				circuit.loc = src.loc
+				circuit.add_fingerprint(user)
+				circuit = null
+		if(2)
+			if(istype(P, /obj/item/weapon/screwdriver) && circuit)
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+				user << "<span class='notice'>You unfasten the circuit board.</span>"
+				state = 1
+				icon_state = "1"
+			if(istype(P, /obj/item/stack/cable_coil))
+				var/obj/item/stack/cable_coil/C = P
+				if(C.get_amount() >= 5)
+					playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
+					user << "<span class='notice'>You start adding cables to the frame...</span>"
+					if(do_after(user, 20/P.toolspeed, target = src))
+						if(C.get_amount() >= 5 && state == 2)
+							C.use(5)
+							user << "<span class='notice'>You add cables to the frame.</span>"
+							state = 3
+							icon_state = "3"
+				else
+					user << "<span class='warning'>You need five lengths of cable to wire the frame!</span>"
+		if(3)
+			if(istype(P, /obj/item/weapon/wirecutters))
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+				user << "<span class='notice'>You remove the cables.</span>"
+				state = 2
+				icon_state = "2"
+				var/obj/item/stack/cable_coil/A = new (loc)
+				A.amount = 5
+				A.add_fingerprint(user)
+
+			if(istype(P, /obj/item/stack/sheet/glass))
+				var/obj/item/stack/sheet/glass/G = P
+				if(G.get_amount() < 2)
+					user << "<span class='warning'>You need two glass sheets to continue construction!</span>"
+					return
+				else
+					playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
+					user << "<span class='notice'>You start to put in the glass panel...</span>"
+					if(do_after(user, 20, target = src))
+						if(G.get_amount() >= 2 && state == 3)
+							G.use(2)
+							user << "<span class='notice'>You put in the glass panel.</span>"
+							state = 4
+							src.icon_state = "4"
+		if(4)
+			if(istype(P, /obj/item/weapon/crowbar))
+				playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
+				user << "<span class='notice'>You remove the glass panel.</span>"
+				state = 3
+				icon_state = "3"
+				var/obj/item/stack/sheet/glass/G = new (loc, 2)
+				G.add_fingerprint(user)
+			if(istype(P, /obj/item/weapon/screwdriver))
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+				user << "<span class='notice'>You connect the monitor.</span>"
+				var/obj/B = new src.circuit.build_path (src.loc, circuit)
+				transfer_fingerprints_to(B)
+				qdel(src)

```

Comment:
the attackby proc needs to return ..() at the end if no special interaction has been triggered.